debezium-core/src/main/java/io/debezium/pipeline/source/AbstractSnapshotChangeEventSource.java

```diff
@@ -63,25 +63,37 @@ public SnapshotResult execute(ChangeEventSourceContext context) throws Interrupt
             throw new RuntimeException(e);
         }
 
+        boolean completed = true;
+
         try {
-            return doExecute(context, ctx, snapshottingTask);
+            snapshotProgressListener.snapshotStarted();
+            SnapshotResult result = doExecute(context, ctx, snapshottingTask);
+
+            return result;
         }
         catch (InterruptedException e) {
+            completed = false;
             LOGGER.warn("Snapshot was interrupted before completion");
-            snapshotProgressListener.snapshotAborted();
             throw e;
         }
         catch (RuntimeException e) {
-            snapshotProgressListener.snapshotAborted();
+            completed = false;
             throw e;
         }
         catch (Throwable t) {
```

Comment:
While you are in the code, do we really wnat to catch Errors? Shouldn't we catch `Exception` only?
Maybe the `catch` could be simplified to a single branch with `instanceof` to handle interruped exception and the rest via encapuslation to `DebeziumException`. In that case the `completed` flag would be unnecessary

Reply:
Yes, catching `Throwable` indeed doesn't sound right. Slightly reworked it, but the flag is still there. I don't think instanceof is really better.